R/sort_linter.R

```diff
@@ -0,0 +1,39 @@
+#' Require usage of `sort()` over `.[order(.)]`
+#'
+#' `sort()` is the dedicated option to sort a list or vector. It is more legible
+#' and around twice as fast as `.[order(.)]`, with the gap in performance
+#' growing with the vector size.
+#'
+#' @evalRd rd_tags("sort_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+sort_linter <- function() {
+  xpath <- "//expr[
```

Comment:
It's better to use `//OP-LEFT-BRACKET` (or possibly `//SYMBOL_FUNCTION_CALL`? the goal is to first narrow down to the least common node; I guess the difference between `SYMBOL_FUNCTION_CALL` and `OP-LEFT-BRACKET` will not matter as much as avoiding `//expr`); see #1358. Try re-expression the path that way. (and sorry that the codebase is still full of misleading examples for you to imitate :) )